### ISSUE TITLE
I Am Very Smart

### ISSUE THREAD

**Issue by Madis0** [Madis0/fabulously-optimized#51]
**Mod name**
I Am Very Smart

**Curseforge link**
https://www.curseforge.com/minecraft/mc-mods/i-am-very-smart

**Modrinth link**

**Other link**

**What it does**
Unlocks all crafting recipes and hides the unlocking toasts

**Why should it be in the modpack**
Unlocking recipes makes no sense for survival Minecraft, only custom maps. This is evident by Bedrock Edition as well, where this feature does not exist.

**Why shouldn't it be in the modpack**
I really don't like the fact that it unlocks the recipes server-side, meaning it _will_ modify all singleplayer worlds you open it in and will not work on nearly any servers

**Categories**

- [ ] Performance optimization
- [ ] Graphics optimization
- [x] New feature
- [ ] Optifine parity
- [ ] Fixes a bug/dependency
- [ ] Replaces an existing mod

**Additional details**
Please vote for this issue: https://gitlab.com/sargunv-mc-mods/i-am-very-smart/-/issues/5

**Comment by TheMadHau5** [comment]
Unlocking recipes does make sense for survival, for example: if I wanted to make a stonecutter, but didn't have any stone lying around and the recipe was not unlocked, I would have to search for the recipe, smelt some cobblestone, and then craft it. The mod makes the experience just a lot more smoother.

**Comment by Madis0** [comment]
> Unlocking recipes does make sense for survival, for example: if I wanted to make a stonecutter, but didn't have any stone lying around and the recipe was not unlocked, I would have to search for the recipe, smelt some cobblestone, and then craft it. The mod makes the experience just a lot more smoother.

Yeah, that's just my bad wording 😁 What I actually meant is that it makes no sense for crafting recipes _to be locked_ in survival.

**Comment by TheMadHau5** [comment]
Ah, I see. Alright then, it seems that we are in agreement that this mod is useful overall. Added in next release?

**Comment by Madis0** [comment]
> Added in next release?

> I really don't like the fact that it unlocks the recipes server-side, meaning it will modify all singleplayer worlds you open it in and will not work on nearly any servers

**Comment by Geefire-temp** [comment]
I would like to say I'm against this. On certain CTM and adventure maps, there are custom crafting recipes locked behind certain requirements. As far as I'm aware, this mod would break those maps by revealing the crafting recipe ahead of time.

**Comment by Madis0** [comment]
@Geefire-temp Fair point. I guess the ideal approach would be to unlock them client-side by default, with the option of explicit relocking by the server. Although that might also relock most of the servers that use 1.8 base and already hide the recipes, haha.

Will see if a clever implementation comes up.
